### ISSUE TITLE
The E-step and stop_gradient

### ISSUE THREAD

**Issue by Germanunkol** [gyang274/capsulesEM#1]
Nice to see there's already an implementation of this!

I just stumbled across [tensorflow's "stop_gradient" function](https://www.tensorflow.org/api_docs/python/tf/stop_gradient). In the examples of where the function might be needed, they mention "The EM algorithm where the M-step should not involve backpropagation through the output of the E-step."

Does this also apply when using the EM algorithm for routing? I don't think I read anything about this in the paper, but then again the paper is very sparse with information about the backpropagation...
Not calculating the gradients for the E-step might considerably speed up training, I believe.
Thoughts?

**Comment by gyang274** [comment]
I agree, there should be no backpropagation through the EM algorithm, other than learning the beta_a and beta_v. Will modify. Thanks.

**Comment by andrewsonga** [comment]
@Germanunkol @gyang274 may I ask, why would not calculating gradients for the E-step might considerably speed up training? If we're unrolling multiple EM iterations, wouldn't blocking gradients to E-step prevent gradients from flowing to earlier EM iterations?
